src/v/raft/group_configuration.cc

```diff
@@ -224,8 +224,10 @@ bool group_configuration::is_allowed_to_request_votes(vnode id) const {
     if (old_it != _old->voters.cend()) {
         return true;
     }
+    // look in learners
+    old_it = std::find(_old->learners.cbegin(), _old->learners.cend(), id);
 
-    return false;
+    return old_it != _old->learners.cend();
```

Comment:
why is the revert necessary?

Reply:
Note on reverted https://github.com/redpanda-data/redpanda/commit/e1c89ecc3531f61b43e39b67beaa8fcdf6d94b79.
The change might caused issues with single replica partitions.

Change introduced in this commit was incorrect as it would lead to situation in which partition might not make progress when learner in old joint configuration is the only node allowed to be a leader. This may be the case as demoting learner in old configuration is a step that comes after it was a voter i.e. a node being a leader demoted to a learner will 'see' that change as a first one. In case of a failure the change may not be able to propagate to other nodes leaving the learner the only node eligible to become a leader.